bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PacketProcessorBase.java

```diff
@@ -85,7 +85,7 @@ protected void sendResponse(int rc, Object response, OpStatsLogger statsLogger)
         final long writeNanos = MathUtils.nowInNano();
 
         final long timeOut = requestProcessor.getWaitTimeoutOnBackpressureMillis();
-        if (timeOut >= 0 && !channel.isWritable()) {
+        if (timeOut > 0 && !channel.isWritable()) {
```

Comment:
timeout is 0 should fail immediately, it is expected.

Reply:
Is it ok when timeout is 0 and blacklistedChannels = Optional.empty() ?